### ISSUE TITLE
Fails to build with Bison 3.6.1

### ISSUE THREAD

**Issue by kaol** [MiniZinc/libminizinc#389]
Debian has updated to Bison version 3.6.1 in unstable release and during its reproducible build checks it turns out that Minizinc 2.4.3 fails to build with it.

The build log is available at https://tests.reproducible-builds.org/debian/rbuild/unstable/amd64/minizinc_2.4.3-1.rbuild.log.gz

**Comment by kaol** [comment]
Debian updated to Bison 3.7 and guess what, the build fails with that too, for a different reason. The generated `tab.cpp` files gained an `#include <...tab.hh>` directive in them and as BISON_TARGET drops them to a different path, the build fails. I compensated for this by adding these to cmake/targets/libminizinc_parser.cmake (but you may want to do something else):
```
execute_process(COMMAND ln -s ${PROJECT_BINARY_DIR}/include/minizinc/parser.tab.hh ${PROJECT_BINARY_DIR}/parser.tab.hh)
execute_process(COMMAND ln -s ${PROJECT_BINARY_DIR}/include/minizinc/support/regex_parser.tab.hh ${PROJECT_BINARY_DIR}/regex_parser.tab.hh)
```

I suppose this might be considered a bug in CMake, too.

**Comment by Dekker1** [comment]
@kaol This issue was resolved in 952ffda0bd23dc21f83d3e3f080ea5b3a414e8e0

The issue is that Bison no longer makes copies of the header file in the source file and instead expect the location of the header to be configured using `%define api.header.include ...`. This does break compatibility with any Bison versions before 3.4.
